src/circuit/floor_planner/single_pass.rs

```diff
@@ -314,6 +370,73 @@ impl<'r, 'a, F: Field, CS: Assignment<F> + 'a> RegionLayouter<F>
     }
 }
 
+pub(crate) struct SimpleTableLayouter<'r, 'a, F: Field, CS: Assignment<F> + 'a> {
+    cs: &'a mut CS,
+    used_columns: &'r [TableColumn],
+    // maps from a fixed column to a pair (default value, vector saying which rows are assigned)
+    pub(crate) default_and_assigned: HashMap<TableColumn, (Option<Option<Assigned<F>>>, Vec<bool>)>,
```

Comment:
Note to self: understand why there is a double `Option<Option<Assigned<F>>>`. e.g. could we have a default value be `Some(None)`?